code/modules/research/nanites/nanite_programs/suppression.dm

```diff
@@ -276,9 +276,12 @@
 	desc = "The nanites synthesize serotonin inside the host's brain, creating an artificial sense of happiness."
 	use_rate = 0.1
 	rogue_types = list(/datum/nanite_program/brain_decay)
-	extra_settings = list(NES_MOOD_MESSAGE)
 	var/message = "HAPPINESS ENHANCEMENT"
 
+/datum/nanite_program/good_mood/register_extra_settings()
+	. = ..()
+	extra_settings[NES_MOOD_MESSAGE] = new /datum/nanite_extra_setting/text("")
+
 /datum/nanite_program/good_mood/set_extra_setting(user, setting)
 	if(setting == NES_MOOD_MESSAGE)
 		var/new_message = stripped_input(user, "Choose the message visible on the mood effect.", NES_MESSAGE, message, MAX_NAME_LEN)
```

Comment:
Seems like this program was never updated to the new extra settings standard for some reason, the set_extra_setting() should be removed and the `message` var should be replaced by directly using `get_value()` on `extra_setting[NES_MOOD_MESSAGE]` instead

Reply:
Will update the rest in a moment